lib/pretty_text.rb

```diff
@@ -406,29 +406,63 @@ def self.convert_vimeo_iframes(doc)
   end
 
   def self.strip_secure_media(doc)
-    doc.css("a[href]").each do |a|
-      if Upload.secure_media_url?(a["href"])
-        target = %w(video audio).include?(a&.parent&.name) ? a.parent : a
-        next if target.to_s.include?("stripped-secure-view-media")
-        width = a.xpath("//*[@width]").attr("width")&.value
-        height = a.xpath("//*[@height]").attr("height")&.value
-        target.add_next_sibling secure_media_placeholder(doc, a['href'], width: width, height: height)
-        target.remove
+    # images inside a lightbox or other link
+    doc.css('a[href]').each do |a|
+      if a.classes.include?('lightbox')
+        img = a.css('img[src]').first
+
+        # we are using the first image from the srcset here so we get the
+        # optimized image instead of the possibly huge original
+        srcset = img.attributes['srcset'].value
+        if Upload.secure_media_url?(a['href'])
+          next if a.to_s.include?('stripped-secure-view-media')
+          url = srcset.split(',').first
+          a.add_next_sibling secure_media_placeholder(doc, url, width: img['width'], height: img['height'])
+          a.remove
+        end
+      else
+        if Upload.secure_media_url?(a['href'])
+          non_image_media = %w(video audio).include?(a&.parent&.name)
+          target = non_image_media ? a.parent : a
+          next if target.to_s.include?('stripped-secure-view-media')
+          width = non_image_media ? nil : a.at_css('img').attr('width')
+          height = non_image_media ? nil : a.at_css('img').attr('height')
+          target.add_next_sibling secure_media_placeholder(doc, a['href'], width: width, height: height)
+          target.remove
+        end
       end
```

Comment:
Both branches if/else has that check `if Upload.secure_media_url?(a['href'])`

I guess we can extract it to guard above so it would unnest it a little bit:

```
doc.css('a[href]').each do |a|
  next if !Upload.secure_media_url?(a['href'])
```